### PR TITLE
[#175798598] Added check payment tests

### PR DESCRIPTION
**Pull request by infantesimone**
#### Description 

The goal of this PR is to add api invocation to validate the _paymentId_ and get payment data (for example  payment amount, email ... ).

#### List of Changes

- fix api spec for _payment/id/action/check_  api;
- add _payment/id/action/check_  api call when the app is loaded;
- save in session storage _idPayment_, _amount_ and (optionally) _email_;
- add integration tests;

#### Motivation and Context

The invocation of the following api (Payment Manager):

  - GET /pp-restapi/v3/payments/:idPayment/actions/check

it is important to validate the _idPayment_ and retrieve the payment information.

#### How Has This Been Tested?

- yarn run test
- yarn start (with local [Payment Manager](https://github.com/pagopa/pm-mock))

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: openapi/api.yml [hunk 1]
```diff
@@ -1958,6 +1958,10 @@ definitions:
     title: PayRequest
   Payment:
     type: object
+    required:
```
Comment (GiovanniMancini) [inline openapi/api.yml:1961]: Maybe we should check if such change has an effect on autogenerated `Payment.decoder()` .. I would expect it to fail if the payment doesn't have the required fields

File: src/__integrations__/pagopaApi.server.integration.test.ts [hunk 2]
```diff
@@ -119,4 +119,121 @@ describe('Payment Manager Client', () => {
...
+      response.fold(
+        _ => fail(),
+        response => response.value?.data?.idPayment,
+      ),
+    ).toEqual(idPayment);
+
+    await stubServerTerminator.terminate();
+  });
+
+  it('should return UnprocessableEntity error (422) when checkPayment endpoint is invoked with idPayment already retrived', async () => {
```
Comment (GiovanniMancini) [inline src/__integrations__/pagopaApi.server.integration.test.ts:159]: Great test! Thank you.

**Review by GiovanniMancini — CHANGES_REQUESTED** [review]
Good test suite! Maybe we should move tests for `check` endpoint in a separate file

File: openapi/api.yml (hunk 1 above)
Comment (infantesimone) [reply openapi/api.yml:1961]: Ok,  these three fields should be required (for the api that io-pay uses). If this is true, is this assumption fine? It's correct  update spec?

**Review by GiovanniMancini — APPROVED** [review]

**Pull request merged**